test/RpcEvmCompatibility.js

```diff
@@ -0,0 +1,275 @@
+    }
+
+    before(async () => {
+        provider = ethers.provider;
+        [signer] = await ethers.getSigners();
+
+        rpcCompatibilityFactory = await ethers.getContractFactory('TestRpcCompatibility', signer);
+        rpcCompatibilityContract = await rpcCompatibilityFactory.deploy();
+        await rpcCompatibilityContract.deployTransaction.wait(network.config.confirmations);
+
+        transferAmount = getRandomInt(maxTransferAmount);
+    });
+
+    it('should support RPC method eth_getLogs', async () => {
+        const newValue = 100;
+        const receipt = await rpcCompatibilityContract.updateValue(newValue).then((tx) => tx.wait());
+        const blockNumber = hexValue(receipt.blockNumber);
+
+        const filter = {
+            fromBlock: blockNumber,
+            toBlock: blockNumber,
+        };
+        const logs = await provider.send('eth_getLogs', [filter]);
+        expect(logs).to.be.an('array');
+        expect(logs.length).to.be.greaterThan(0);
+    });
+
+    describe('rpc get transaction and blockByHash methods', () => {
+        let tx;
+
+        before(async () => {
+            tx = await signer.sendTransaction({
+                to: signer.address,
+                value: transferAmount,
+            });
+            await tx.wait();
+        });
+
+        it('should support RPC method eth_getTransactionReceipt', async () => {
+            const receipt = await provider.send('eth_getTransactionReceipt', [tx.hash]);
+
+            expect(receipt).to.be.an('object');
+            expect(parseInt(receipt.blockNumber, 16)).to.be.a('number');
+            expect(getAddress(receipt.to)).to.equal(signer.address);
+        });
+
+        it('should support RPC method eth_getTransactionByHash', async () => {
+            const txInfo = await provider.send('eth_getTransactionByHash', [tx.hash]);
+
+            expect(txInfo).to.be.an('object');
+            expect(getAddress(txInfo.to)).to.equal(signer.address);
+            expect(parseInt(txInfo.value, 16).toString()).to.equal(transferAmount.toString());
+        });
+
+        it('should support RPC method eth_getBlockByHash', async () => {
+            const receipt = await provider.send('eth_getTransactionReceipt', [tx.hash]);
+            const blockHash = receipt.blockHash;
+            const block = await provider.send('eth_getBlockByHash', [blockHash, true]);
+
+            expect(block).to.be.an('object');
+            expect(block.hash).to.equal(blockHash);
+            expect(parseInt(block.number, 16)).to.be.a('number');
+            expect(parseInt(block.timestamp, 16)).to.be.a('number');
+            checkBlockTimeStamp(parseInt(block.timestamp, 16));
+            expect(block.transactions).to.be.an('array');
+        });
+    });
+
+    it('should support RPC method eth_getBlockByNumber', async () => {
+        const block = await provider.send('eth_getBlockByNumber', ['latest', true]);
```

Comment:
should also test retrieving block by number (say block 1), safe, and finalized block tags. Same for eth_getLogs